app/models/foreman_azure_rm/azure_rm.rb

```diff
@@ -119,33 +129,29 @@ def available_subnets
       subnets
     end
 
-    def virtual_networks(region = nil)
-      if region.nil?
-        sdk.vnets
-      else
-        stripped_region = region.gsub(/\s+/, '').downcase
-        sdk.vnets.select { |vnet| vnet.location == stripped_region }
-      end
+    def virtual_networks
+      @virtual_networks ||= sdk.vnets.select { |vnet| vnet.location == region }
     end
 
-    def subnets(region = nil)
-      stripped_region = region.gsub(/\s+/, '').downcase
-      vnets   = virtual_networks(stripped_region)
+    def subnets
+      vnets   = virtual_networks
       subnets = []
       vnets.each do |vnet|
         subnets.concat(sdk.subnets(vnet.resource_group, vnet.name))
       end
       subnets
     end
```

Comment:
Currently method `available_subnets` internally calls to `subnets` then why not to it alias_method of `subnets`?

`alias_method :available_subnets, :subnets`